### ISSUE TITLE
Full Disk Access in CI

### ISSUE THREAD

**Issue by vashworth** [flutter/flutter#132674]
I was working on a [recipes update](https://flutter-review.googlesource.com/c/recipes/+/49240) that involved getting and updating contents of a directory on the host machine.

However, it would fail on some bots and pass on other bots. This is due to some bots having Full Disk Access and some not. For my change to work, Full Disk Access is needed.

Example:
flutter-devicelab-mac-6 doesn't have access:
https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_ios%20spell_check_test/3685/overview 

flutter-devicelab-mac-24 has access:
https://ci.chromium.org/ui/p/flutter/builders/prod/Mac_ios%20spell_check_test/3686/overview

This fix I added is needed to run our tests on physical iOS 17 devices. We can't update the devicelab until my recipes update is added.

Can we give all bots Full Disk Access? Or is there possibly a way to select bots with Full Disk Access when running tests on physical iOS 17 devices or "xcode_debug" tests (which force test the Xcode automation workflow).

**Comment by godofredoc** [comment]
Only device lab devices have full access to disk and we are working on removing the need for it. Chromium hosted machines do not have access to the full disk because of security concerns.

**Comment by vashworth** [comment]
@godofredoc Do you know timeline on devicelab device getting full disk access removed? Is that already in progress? In the examples I listed, flutter-devicelab-mac-6 doesn't have access and flutter-devicelab-mac-24 does, and I believe both of those are devicelab bots?

TLDR of context: Due to new workflows, we won't be able to run tests in CI on physical iOS 17 devices without Full Disk Access.

Context: Xcode 15 broke `flutter run` on physical iOS 17 devices, so we had to create a new workflow to fix it. The new workflow involved automating Xcode via OSA scripting. However, OSA scripting requires permission to be run ([see example here](https://github.com/flutter/flutter/pull/131865#issuecomment-1666234593)). Since there's no user in CI to give permission, I added a [change to recipes](https://flutter-review.googlesource.com/c/recipes/+/49240) that automates the giving of permission but it requires Full Disk Access. And so, basically we can't run tests in CI on physical iOS 17 devices without Full Disk Access.

**Comment by godofredoc** [comment]
If this is temporary we can add a new dimension for the bots with full disk access so the test will use only those machines.

**Comment by vashworth** [comment]
The hope is for it to be temporary, but we don't have a timeline on permanent fix. A new dimension would be great!

**Comment by godofredoc** [comment]
@yusuf-goog how can we add a new dimension for test beds that have full disk access?

**Comment by yusuf-goog** [comment]
@vashworth currently I think https://g3doc.corp.google.com/company/teams/flutter/infrastructure/devicelab/macos_host.md#1-disable-readonly-filesystem might be responsible for the discrepency on the 2 devicelab hosts.

If you have a concrete check, we could confirm if this is the cause.

**Comment by vashworth** [comment]
I believe you can do `csrutil status` and it should tell you if it's enabled or not

**Comment by yusuf-goog** [comment]
if this solves the issue, i'll run a sweep across all our devices and see if we have any other straglers.

**Comment by vashworth** [comment]
I'm pretty confident fixing https://github.com/flutter/flutter/issues/132700 should fix my issue. I just discovered CircleCI does something very similar to what I'm doing, and their [documentation](https://circleci.com/docs/testing-macos/) confirms System Integrity Protection needs to be disabled (or Full Disk Access given)

**Comment by yusuf-goog** [comment]
opened for other macs: https://github.com/flutter/flutter/issues/132705

**Comment by yusuf-goog** [comment]
@vashworth mac-6 has been fixed

**Comment by vashworth** [comment]
Confirmed that access has been given and fixed my issue: https://chromium-swarm.appspot.com/task?id=64227a92e3ae1e10
